src/commcare_cloud/ansible/roles/nginx/files/dynamic_cfg_change_nginx

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env bash
+
+usage() {
+    declare -r script_name=$(basename "$0")
+    echo """
+Usage:
+"${script_name}" [option] <name>
+
+Option:
+    --overall_restore <INT>
+    --incremental_restore <INT>
+    --submission <INT>
+    --app_download <INT>
+    --multimedia <INT>
+    --default_server <INT>
+"""
+}
+
+main() {
+    if [[ $# -eq 0 ]]; then
+        usage
+        exit 1
+    fi
+
+    OPTIONS=$( getopt -a -o '' -l help,overall_restore:,incremental_restore:,submission:,app_download:,multimedia:,default_server: -- "$@" )
+
+    if [[ $? -ne 0 ]] ; then
+        usage ; exit 1
+    fi
+
+    eval set -- "$OPTIONS"
+
+    while true
+    do
+        case $1 in
+        
+        --help) usage; exit 0 ;;
+
+        --overall_restore) overall_restore=$2; shift 2;;
+        
+        --incremental_restore) incremental_restore=$2; shift 2;;
+        
+        --submission) submission=$2; shift 2;;
+
+        --app_download) app_download=$2; shift 2;;
+
+        --multimedia) multimedia=$2; shift 2;;
+
+        --default_server) default_server=$2; shift 2;;
+
+        --) shift ; break ;;
+        esac
+    done
+
+
+    cfg_file='/etc/nginx/sites-available/icds_cas_commcare'
+
+    if [[ "$incremental_restore" ]]; then
+            old_incremental_restore=`grep -E '^limit_req_zone.*incremental_restore' $cfg_file`
+            replace_incremental_restore="limit_req_zone \$incremental_sync zone=incremental_restore:10m rate=${incremental_restore}r/s;"
+            sed -i "s[$old_incremental_restore[$replace_incremental_restore[g" $cfg_file
+    fi
+
+    if [[ "$overall_restore" ]]; then
+            old_overall_restore=`grep -E '^limit_req_zone.*overall_restore' $cfg_file`
+            replace_overall_restore="limit_req_zone \$limit_bucket zone=overall_restore:10m rate=${overall_restore}r/s;"
+            sed -i "s[$old_overall_restore[$replace_overall_restore[g" $cfg_file
```

Comment:
could extract these 3 lines to a function
```
function change_limits() {
    $cfg_file=$1
    $zone=$2
    $var=$3
    $limit=$4
    old_config=`grep -E '^limit_req_zone.*${zone}' ${cfg_file}`
    new_config="limit_req_zone \$${var} zone=${zone}:10m rate=${limit}r/s;"
    sed -i "s[$old_config[$new_config[g" ${cfg_file}
}
change_limits "$cfg_file" "overall_restore" "limit_bucket" "$overall_restore"
```

Reply:
Updated